sphinx/ext/autosummary/generate.py

```diff
@@ -208,7 +209,7 @@ def scan(self, imported_members: bool) -> List[str]:
                 if inspect.ismodule(value):
                     imported = True
                 elif safe_getattr(value, '__module__') != self.object.__name__:
-                    imported = True
+                    imported = objtype != 'data' or ('', name) not in attr_docs
```

Comment:
Is this condition related to the `__module__`? I think this check should be separated to another condition.

Reply:
In autodoc, global variables having doc-comment are documented without `:imported-members:` option even if their `__module__` value are different because they're considered as "locally defined variables".

```
class Foo:
    pass


#: a global variable having different __module__
var1 = int

#: a global variable having same __module__
var2 = Foo
```

So I think it's better to check the member has doc-comment or not first to determine `imported` flag.

```
if ('', name) in attr_docs:
  imported = False
elif inspect.ismodule(value):
  ...
```